plugins/module_utils/template.py

```diff
@@ -135,3 +140,72 @@ def get_vlan_pool_name(self, vlan_pool_uuid):
         kv_list = [KVPair("uuid", vlan_pool_uuid)]
         match = self.get_object_by_key_value_pairs("VLAN Pool", existing_vlan_pools, kv_list, fail_module=True)
         return match.details.get("name")
+
+    def get_route_map(self, attr_name, tenant_id, tenant_name, route_map, route_map_objects):
+        """
+        Retrieves the details of a specific route map object based on the provided attributes.
+
+        Args:
+            attr_name: The attribute name for error messaging. -> Str
+            tenant_id: The ID of the tenant. -> Str
+            tenant_name: The name of the tenant. -> Str
+            route_map: The name of the route map. -> Str
+            route_map_objects: The list of route map objects to search from. -> List
+
+        Returns:
+            The details of the route map object if found, otherwise an empty dictionary. -> Dict
+        """
+        if route_map and tenant_id and route_map_objects:
+            route_map_object = self.get_object_from_list(
+                route_map_objects,
+                [KVPair("name", route_map), KVPair("tenantId", tenant_id)],
+            )
+            if route_map_object[0]:
+                return route_map_object[0].details
+            else:
+                self.mso.fail_json(msg="Provided Route Map {0}: {1} with the tenant: {2} not found.".format(attr_name, route_map, tenant_name))
+        else:
+            return {}
+
+    def get_vrf_object(self, vrf_dict, tenant_id, templates_objects_path):
+        """
+        Get VRF object based on provided parameters.
+        :param vrf_dict: Dictionary containing VRF details. -> Dict
+        :param tenant_id: Id of the tenant. -> Str
+        :param templates_objects_path: Path to the templates objects. -> Str
+        :return: VRF object if found, otherwise fail with an error message. -> Dict
+        """
+        vrf_params = {"type": "vrf", "tenant-id": tenant_id, "include-common": "true"}
+        vrf_path = self.generate_api_endpoint(templates_objects_path, **vrf_params)
+        vrf_objects = self.mso.query_objs(vrf_path)
+        vrf_kv_list = [
+            KVPair("name", vrf_dict.get("name")),
+            KVPair("templateName", vrf_dict.get("template")),
+            KVPair("schemaName", vrf_dict.get("schema")),
+            KVPair("tenantId", tenant_id),
+        ]
+
+        vrf_object = self.get_object_from_list(vrf_objects, vrf_kv_list)
+
+        if vrf_object[0]:
+            return vrf_object[0]
+        else:
+            self.mso.fail_json(msg="Provided VRF {0} not found.".format(vrf_dict.get("name")))
+
+    @staticmethod
+    def generate_api_endpoint(path, **kwargs):
```

Comment:
is template the correct place for this function or should it be a generic function in module utils?